bc_obps/reporting/service/report_emission_allocation_service.py

```diff
@@ -129,14 +143,12 @@ def save_emission_allocation_data(
                     continue
 
                 report_emission_allocation_record, _ = ReportProductEmissionAllocation.objects.update_or_create(
+                    report_emission_allocation=report_emission_allocation,
```

Comment:
I created a card for this service: https://github.com/bcgov/cas-reporting/issues/694 but we'll leave it like this for now, the issues are not introduced by your PR